Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_ctf.nut

```diff
@@ -151,50 +160,76 @@ void function CreateFlags()
 		{
 			file.militiaFlagSpawn = base
 			file.militiaFlag = flag
-			file.militiaFlagReturnTrigger = returnTrigger
 			
 			SetGlobalNetEnt( "milFlag", file.militiaFlag )
 			SetGlobalNetEnt( "milFlagHome", file.militiaFlagSpawn )
 		}
 	}
 	
-	// reset the flag states, prevents issues where flag is home but doesnt think it's home when halftime goes
 	SetFlagStateForTeam( TEAM_MILITIA, eFlagState.None )
 	SetFlagStateForTeam( TEAM_IMC, eFlagState.None )
-	
-	foreach ( entity player in GetPlayerArray() )
-		CTFInitPlayer( player )
 }
 
 void function RemoveFlags()
 {
-	// destroy all the flag related things
 	if ( IsValid( file.imcFlagSpawn ) )
 	{
+		PlayFX( $"P_phase_shift_main", file.imcFlagSpawn.GetOrigin() )
 		file.imcFlagSpawn.Destroy()
 		file.imcFlag.Destroy()
-		file.imcFlagReturnTrigger.Destroy()
 	}
+	
 	if ( IsValid( file.militiaFlagSpawn ) )
 	{
+		PlayFX( $"P_phase_shift_main", file.militiaFlagSpawn.GetOrigin() )
 		file.militiaFlagSpawn.Destroy()
 		file.militiaFlag.Destroy()
-		file.militiaFlagReturnTrigger.Destroy()
 	}
-
-	// unsure if this is needed, since the flags are destroyed? idk
+	
 	SetFlagStateForTeam( TEAM_MILITIA, eFlagState.None )
 	SetFlagStateForTeam( TEAM_IMC, eFlagState.None )
 }
 
 void function RateSpawnpoints_CTF( int checkClass, array<entity> spawnpoints, int team, entity player ) 
 {
-	RateSpawnpoints_SpawnZones( checkClass, spawnpoints, team, player )
+	vector allyFlagSpot
+	vector enemyFlagSpot
+	foreach ( entity spawn in GetEntArrayByClass_Expensive( "info_spawnpoint_flag" ) )
+	{
+		if( spawn.GetTeam() == team )
+			allyFlagSpot = spawn.GetOrigin()
+		else
+			enemyFlagSpot = spawn.GetOrigin()
+	}
+	
+	foreach ( entity spawn in spawnpoints )
+	{
+		float rating = 0.0
+		float allyFlagDistance = Distance2D( spawn.GetOrigin(), allyFlagSpot )
+		float enemyFlagDistance = Distance2D( spawn.GetOrigin(), enemyFlagSpot )
+		
+		if( enemyFlagDistance > allyFlagDistance )
+		{
+			rating += spawn.NearbyAllyScore( team, "ai" )
+			rating += spawn.NearbyAllyScore( team, "titan" )
+			rating += spawn.NearbyAllyScore( team, "pilot" )
+			
+			rating += spawn.NearbyEnemyScore( team, "ai" )
+			rating += spawn.NearbyEnemyScore( team, "titan" )
+			rating += spawn.NearbyEnemyScore( team, "pilot" )
+		
+			rating = rating / allyFlagDistance
+		}
+
+		if ( spawn == player.p.lastSpawnPoint )
+			rating += GetConVarFloat( "spawnpoint_last_spawn_rating" )
```

Comment:
What's the default value for this convar? Should this be inverted to `-=` (We presumably don't want to spawn the player in the same place over and over, so if the default value is >0 it should be inverted)

Reply:
Default value is `-4` so inverting would actually do what you just mentioned.